test/src/unittests/tonal/test_pitchsaliencefunctionpeaks.py

```diff
@@ -0,0 +1,160 @@
+
+    def testRegression3PeaksHighBinResolution(self):
+        # Checks on config. values binResolution being too high for the input in question
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction(binResolution=2400)(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(values, [])
+
+    def testRegressionBadMaxFreq(self):
+        # Checks on config. values maxFrequency being too low for a given set of spectral peaks
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks(maxFrequency=50)(calculatedPitchSalience))
+
+    def testRegressionBadMinFreq(self):
+        # Checks on config. values minFrequency being too high for a given set of spectral peaks
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks, mag_speaks)
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks(minFrequency=20000)(calculatedPitchSalience))
+
+    def testRegressionBadRefFreq(self):
+        # Checks on config. values referenceFrequency being too high for a given set of spectral peaks
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks, mag_speaks)
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks(referenceFrequency=20000)(calculatedPitchSalience))
+
+    def testRegression3PeaksHw0UnharmonicInput(self):
+        # As in testRegression3Peaks but with Harmonic Weight = 0, non multiple freq. inputs        
+        freq_speaks = [55, 100, 340]
+        mag_speaks = [1, 1, 1]
+        expectedBins =  [0., 103., 315.]
+        expectedValues = [1., 1., 1.]
+        calculatedPitchSalience = PitchSalienceFunction(harmonicWeight=0)(freq_speaks, mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertAlmostEqualVectorFixedPrecision(bins, expectedBins, 6)
+        self.assertAlmostEqualVectorFixedPrecision(values, expectedValues, 6)
+    
+    def testRegression3PeaksHw0HarmonicInput(self):
+        # As in testRegression3Peaks but with Harmonic Weight = 0, multiple freq. inputs        
+        freq_speaks = [110, 220]
+        mag_speaks = [1, 1]
+        expectedBins =  [120., 240.]
+        expectedValues = [1., 1.]
+        calculatedPitchSalience = PitchSalienceFunction(harmonicWeight=0)(freq_speaks, mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertAlmostEqualVectorFixedPrecision(bins, expectedBins, 6)
```

Comment:
Use assertEqualVector for expectedBins.

Reply:
I have to use assertAlmostEqualVectorFixedPrecision() becuase the value returned is float vector for bins.
assertEqualVector doesnt work here.